src/components/sessions/SessionControl.tsx

```diff
@@ -152,8 +172,30 @@ const SessionControl = () => {
   };
 
   useEffect(() => {
-    if (sessionState === "START" && timeInSeconds === -1) {
-      onStopSession();
+    let interval: NodeJS.Timeout;
+
+    const checkPosture = () => {
+      const posture = {
+        good_posture: currentPosture === "good",
+        recorded_at: new Date().toISOString(),
+      };
+      setPostureData((prevState) => [...prevState, posture]);
+    };
+
+    if (sessionState === "START" && timerState === "RUNNING") {
+      interval = setInterval(() => {
+        checkPosture();
+      }, 5000);
+    }
+
+    return () => {
+      clearInterval(interval);
+    };
+  }, [sessionState, timerState, currentPosture, postureData]);
+
+  useEffect(() => {
+    if (sessionState === "START" && timeInSeconds === 0) {
+      handleEndSession();
```

Comment:
can you disabled the start button if the timeInSeconds is 0 instead of this validation?

Reply:
In this case, there is no start button since the timer will be running, the Start button is not visible, in this case the button is Stop Tracking but as soon as the timer gets to 0, we get the session summary which is coming on the handleEndSession.